### PR TITLE
Auth consistency changes

### PR DESCRIPTION
**Pull request by hamchapman**
### What?

Bring auth flow in line with pusher-platform-node

### Why?

Consistency. Glory.

----

- [x] README updated if you changed the API?

----

CC @pusher/sigsdk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,10 +39,11 @@ class AuthController < ActionController::Base
   before_action :authenticate_user!
 
   def auth
-    render json: pusher.authenticate_with_request(
+    auth_data = pusher.authenticate_with_request(
       request,
       { user_id: current_user.id }
     )
+    render json: auth_data.body, status: auth_data.status
```
Comment (pl) [inline README.md]: Would be good to add headers here.

File: lib/pusher-platform/authentication_response.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      "PusherPlatform::AuthenticationResponse - status: #{@status} body: #{@body.to_json}"
+    end
+
+    def as_json(options = {})
+      json = {
+        status: @status,
+        headers: @headers,
+        body: @body
+      }
+      json
```
Comment (pl) [inline lib/pusher-platform/authentication_response.rb]: Nitpicking – not sure why you can't just return the object without assignment.

Comment (hamchapman) [reply lib/pusher-platform/authentication_response.rb]: Good spot - think it was split from what I was doing some logging but forgot to kill it. Will do now 👍

File: README.md (hunk 1 above)
Comment (hamchapman) [reply README.md]: Will do

**Pull request merged**